### PR TITLE
maintenance: Emit the same warnings across the toolstack and remove a few dozen of them

### PR DESCRIPTION
**Pull request by psafont**
This was hiding a large amount of compiler warnings, especially in xenopsd.

Now the flags are controlled by the dune file at the root of the project.

There are some exceptions to this rule for heavy ppx and autogenerated code where the warning are not easily avoidable and aren't useful.

### CODE REVIEW

File: ocaml/tests/alerts/test_alert_certificate_check.ml [hunk 1]
```diff
@@ -26,24 +26,22 @@ let good_samples =
 
 let expiring_samples =
   [
-    ("20200302T02:00:00Z", 30, Api_messages.host_server_certificate_expiring_30)
-  ; ("20200301T02:00:00Z", 29, Api_messages.host_server_certificate_expiring_30)
-  ; ("20200215T02:00:00Z", 14, Api_messages.host_server_certificate_expiring_14)
-  ; ("20200208T02:00:00Z", 7, Api_messages.host_server_certificate_expiring_07)
-  ; ("20200201T02:00:00Z", 0, Api_messages.host_server_certificate_expiring_07)
+    ("20200302T02:00:00Z", Api_messages.host_server_certificate_expiring_30)
```
Comment (edwintorok) [inline ocaml/tests/alerts/test_alert_certificate_check.ml:29]: are we sure we want to remove these numbers here?

Comment (edwintorok) [reply ocaml/tests/alerts/test_alert_certificate_check.ml:29]: Ah I see, it wasn't used in the formatted message (it probably changed).

Comment (psafont) [reply ocaml/tests/alerts/test_alert_certificate_check.ml:29]: The numbers are already encoded in the name of the message and not used to generate the message, so pretty sure ;)

**Review by lindig — APPROVED** [review]
Looking over it, it looks good to me. Mostly prefixing unused parameters with `_` and some dead code removed.

**Review by robhoes — APPROVED** [review]

**Pull request merged**